Classes/Query/Builder.php

```diff
@@ -0,0 +1,139 @@
+
+/***************************************************************
+ *  Copyright notice
+ *
+ *  (c) 2009-2012 Ingo Renner <ingo@typo3.org>
+ *  All rights reserved
+ *
+ *  This script is part of the TYPO3 project. The TYPO3 project is
+ *  free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  The GNU General Public License can be found at
+ *  http://www.gnu.org/copyleft/gpl.html.
+ *
+ *  This script is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  This copyright notice MUST APPEAR in all copies of the script!
+ ***************************************************************/
+
+use ApacheSolrForTypo3\Solr\Site;
+use ApacheSolrForTypo3\Solr\Util;
+use ApacheSolrForTypo3\Solrmlt\Configuration\PluginConfiguration;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
+use TYPO3\CMS\Frontend\Plugin\AbstractPlugin;
+
+/**
+ * Builder class responsible to build an initialized more like this query.
+ *
+ * @author Timo Schmidt
+ * @package    TYPO3
+ * @subpackage    solr
+ */
+class Builder
+{
+    /**
+     * @return \ApacheSolrForTypo3\Solrmlt\Query\Query
+     */
+    protected function getMoreLikeThisQuery()
+    {
+        return GeneralUtility::makeInstance('ApacheSolrForTypo3\\Solrmlt\\Query\\Query');
+    }
+
+    /**
+     * @param PluginConfiguration $pluginConfiguration
+     * @param TypoScriptFrontendController $TSFE
+     * @return \ApacheSolrForTypo3\Solrmlt\Query\Query
+     */
+    public function build(PluginConfiguration $pluginConfiguration, TypoScriptFrontendController $TSFE)
+    {
+        /** @var $query \ApacheSolrForTypo3\Solrmlt\Query\Query */
+        $query = $this->getMoreLikeThisQuery();
+        $query->setQueryFields(array('id,title', 'score'));
+
+        $query = $this->applyFrontedRestrictions($query, $TSFE);
+        $query = $this->applyPluginConfiguration($query, $pluginConfiguration);
+        $query = $this->applyQueryString($query, $pluginConfiguration, $TSFE);
+
+        return $query;
+    }
+
+    /**
+     * @param Query $query
+     * @param PluginConfiguration $pluginConfiguration
+     * @param $TSFE
+     * @return Query
+     */
+    protected function applyQueryString(Query $query, PluginConfiguration $pluginConfiguration, $TSFE)
+    {
+        switch ($pluginConfiguration->getQueryStringCreationType()) {
+            case 'pageid':
+                $pageId = $TSFE->id;
+                $typeNum = $TSFE->type;
+                $language = $TSFE->sys_language_uid;
+
+                // what do we need to use here?
+                $accessGroups = '0';
```

Comment:
should be c:0 I believe(?)
Ideally should use the FE user's groups, but then we'd have caching issues. So only public/anonymous content is ok.